stable/artifactory-ha/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,27 @@
+{{- if .Values.metrics.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "artifactory-ha.fullname" . }}
+  labels:
+    app: {{ template "artifactory-ha.name" . }}
+    chart: {{ template "artifactory-ha.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    {{- with .Values.metrics.serviceMonitor.labels }}
+{{ toYaml . | indent 4 }}
+    {{- end }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "artifactory-ha.name" . }}
+      component: metrics
+      release: {{ .Release.Name }}
+  endpoints:
+  - port: tcp-metrics
+    path: "/metrics"
```

Comment:
I'm not very familiar with this object, but shouldn't this be in the values.yaml to allow changing it?

Reply:
This is the default path for the Fluentd Prometheus output plugin.  We can make this configurable but we'd also need to use that same configuration value in the Fluentd configuration file.

https://github.com/fluent/fluent-plugin-prometheus

Reply:
Currently, is it "hard-coded" in the fluetnd configuration as well? If so, I think it needs to move to the values.yaml rather than duplicating this value.

Reply:
The port and path can be configured in Fluentd by editing this section of the configuration file:

```
# input plugin that exports metrics
<source>
  @type prometheus
  port 24231
  metrics_path /metrics
</source>
```

The best way I can think of to get configuration values from the Helm `values.yaml` into this Fluentd config file would be using environment variables similar to how we are setting the `JFROG_LOGS_DIR`.  Now that you bring this up I definitely think it's worth keeping them in sync and configurable.  Do you think this is a good way to do it?

Reply:
This config file uses `tpl` anyway, so you can simply use values as they are. e.g. `.Values.metrics.port`